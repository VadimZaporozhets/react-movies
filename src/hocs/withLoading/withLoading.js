import React from 'react';
import CircularProgress from '@material-ui/core/CircularProgress';
import { withStyles } from '@material-ui/core';
import { bool, object } from 'prop-types';

import { styles } from './withLoading.styles';

export const withLoading = Component => {
    const WithLoadingComponent = ({ loading, classes, ...props }) => (
        <>
            {loading ? (
                <CircularProgress className={classes.progress} />
            ) : (
                <Component {...props} />
            )}
        </>
    );

    WithLoadingComponent.propTypes = {
        loading: bool.isRequired,
        classes: object.isRequired
    };

    return withStyles(styles)(WithLoadingComponent);
};
