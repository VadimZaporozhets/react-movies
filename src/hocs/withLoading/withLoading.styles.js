export const styles = () => ({
    progress: {
        margin: 'auto',
        display: 'block'
    }
});
