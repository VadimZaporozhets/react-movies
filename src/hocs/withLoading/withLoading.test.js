import React from 'react';
import { shallow } from 'enzyme';

import { withLoading } from './withLoading';

describe('withLoading', () => {
    it('should render loading indicator when loading flag activated', () => {
        const props = {
            loading: true
        };
        const WithLoadingComponent = withLoading(<div />);
        const wrapper = shallow(<WithLoadingComponent {...props} />);

        expect(wrapper).toMatchSnapshot();
    });

    it('should render component when loading flag is inactive', () => {
        const props = {
            loading: false
        };
        const WithLoadingComponent = withLoading(<div />);
        const wrapper = shallow(<WithLoadingComponent {...props} />);

        expect(wrapper).toMatchSnapshot();
    });
});
