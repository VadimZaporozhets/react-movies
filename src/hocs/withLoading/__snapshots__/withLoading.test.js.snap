// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withLoading should render component when loading flag is inactive 1`] = `
<WithLoadingComponent
  classes={
    Object {
      "progress": "WithLoadingComponent-progress-2",
    }
  }
  loading={false}
/>
`;

exports[`withLoading should render loading indicator when loading flag activated 1`] = `
<WithLoadingComponent
  classes={
    Object {
      "progress": "WithLoadingComponent-progress-1",
    }
  }
  loading={true}
/>
`;
